components/gfx/text/glyph.rs

```diff
@@ -547,6 +547,27 @@ impl<'a> GlyphStore {
     }
 
     #[inline]
+    pub fn range_index_of_advance(&self, range: &Range<ByteIndex>, advance: Au, extra_word_spacing: Au) -> (usize, Au) {
+        let mut index = 0;
+        let mut current_advance = Au(0);
+        self.iter_glyphs_for_byte_range(range)
+            .take_while(|glyph| {
+                if glyph.char_is_space() {
+                    current_advance += glyph.advance() + extra_word_spacing
+                } else {
+                    current_advance += glyph.advance()
+                }
+                if current_advance > advance {
+                    false
+                } else {
+                    index += 1;
+                    true
+                }
+            }).collect::<Vec<_>>();
```

Comment:
It looks like you're collecting this into a vector and then throwing it away.

I recommend just using a for loop:

    for glyph in self.iter_glyphs_for_byte_range(range) {
        ...
    }